plugins/woocommerce-blocks/tests/e2e/tests/local-pickup/local-pickup.merchant.block_theme.spec.ts

```diff
@@ -0,0 +1,292 @@
+/**
+ * External dependencies
+ */
+import { test, expect } from '@woocommerce/e2e-playwright-utils';
+
+/**
+ * Internal dependencies
+ */
+import { utilsLocalPickup as utils } from './utils.local-pickup';
+
+test.describe( 'Merchant → Local Pickup Settings', () => {
+	test.beforeEach( async ( { admin, page } ) => {
+		await utils.clearLocations( admin, page );
+		await utils.removeCostForLocalPickup( admin, page );
+		await utils.enableLocalPickup( admin, page );
+	} );
+
+	test( 'Renders without crashing, and settings persist', async ( {
+		admin,
+		page,
+	} ) => {
+		await admin.visitAdminPage(
+			'admin.php',
+			'page=wc-settings&tab=shipping&section=pickup_location'
+		);
+		await page.waitForSelector( '#local-pickup-settings' );
+		expect( page.locator( '#local-pickup-settings' ) ).not.toBeNull();
+		await expect(
+			page.locator( '#inspector-checkbox-control-0' )
+		).toBeChecked();
+	} );
+
+	test.describe( 'Core Settings', () => {
+		test.beforeEach( async ( { admin, page } ) => {
+			await utils.clearLocations( admin, page );
+			await utils.removeCostForLocalPickup( admin, page );
+			await utils.enableLocalPickup( admin, page );
+		} );
+
+		test( 'Shows the correct shipping options depending on whether Local Pickup is enabled', async ( {
+			admin,
+			page,
+		} ) => {
+			await utils.disableLocalPickup( admin, page );
+
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=pickup_location'
+			);
+
+			await expect(
+				page.locator( '#inspector-checkbox-control-0' )
+			).not.toBeChecked();
+
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=options'
+			);
+
+			expect(
+				page.locator( '#woocommerce_shipping_cost_requires_address' )
+			).not.toBeNull();
+
+			await utils.enableLocalPickup( admin, page );
+
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=options'
+			);
+
+			expect(
+				page.locator( '#woocommerce_shipping_cost_requires_address' )
+			).not.toBeNull();
+		} );
+	} );
+
+	test.describe( 'Global Settings', () => {
+		test.beforeEach( async ( { admin, page } ) => {
+			await utils.clearLocations( admin, page );
+			await utils.removeCostForLocalPickup( admin, page );
+			await utils.enableLocalPickup( admin, page );
+		} );
+
+		test( 'Allows toggling of enabled state', async ( { admin, page } ) => {
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=pickup_location'
+			);
+
+			const text = 'Enable local pickup';
+			const xpath = `//label[text()='${ text }']/preceding-sibling::span`;
+			const checkedState = page
+				.locator( xpath )
+				.locator( 'svg' )
+				.isVisible();
+
+			await page.getByText( text ).click();
+			await utils.savelocalPickupSettings( admin, page );
+
+			const newCheckedState = page
+				.locator( xpath )
+				.locator( 'svg' )
+				.isVisible();
+
+			expect( newCheckedState ).not.toBe( checkedState );
```

Comment:
```suggestion
			await expect(
				page.getByLabel( 'Enable local pickup' )
			).toBeChecked();

			await page.getByLabel( 'Enable local pickup' ).click();

			await expect(
				page.getByLabel( 'Enable local pickup' )
			).not.toBeChecked();
```

Rather than relying on XPath, what are your thoughts on directly toggling the checkbox and checking if the checkbox is checked respectively unchecked?

Reply:
I've applied these changes locally and will push them soon.